docs: add status explanation to readme (#475)
open-cluster-management-io/config-policy-controller: README.md

    - lastTimestamp: "2026-05-28T16:18:34.733058Z"
      message: NonCompliant; violation - configmaps [example-configmap] found but not as specified in namespace default
  lastEvaluated: "2026-05-28T16:23:14Z"
  lastEvaluatedGeneration: 2
  relatedObjects:
    - compliant: NonCompliant
      object:
        apiVersion: v1
        kind: ConfigMap
        metadata:
          name: example-configmap
          namespace: default
      properties:
        createdByPolicy: false
        diff: |
          --- default/example-configmap : existing
          +++ default/example-configmap : updated
          @@ -1,9 +1,9 @@
           apiVersion: v1
           data:
             env: dev
          -  purpose: load-testing
          +  purpose: load-test
           kind: ConfigMap
           metadata:
             creationTimestamp: "2026-05-28T16:17:54Z"
             name: example-configmap
             namespace: default
        uid: f706c4be-5b68-4592-a81e-d9be6d8fd25b
      reason: Resource found but does not match
      cluster: local-cluster
```

##### Status field descriptions

| Field | Description |
| ---- | ---- |

> | `compliant` | Overall compliance state: `Compliant` (all templates match), `NonCompliant` (one or more templates do not match), `Pending` (evaluation in progress), or `Terminating` (policy is being deleted) |

Reviewer: `Pending` is an odd case here: it is allowed in the CRD validation, but I don't think the controller should ever use it. I think it's only supposed to be visible at the Policy level, not the template level. I suggest we just don't mention it here.

| `lastEvaluated` | ISO 8601 timestamp of the most recent evaluation |
| `lastEvaluatedGeneration` | The generation of the ConfigurationPolicy resource at the last evaluation |
| `compliancyDetails` | Array with one entry per object-template, showing compliance state and violation details |

> | `relatedObjects` | List of all Kubernetes objects matched by the policy templates |

Reviewer: Nit: call them both "arrays", or both "lists".

| `history` | Timestamped messages showing the recent compliance state changes |

##### Related object properties

| Field | Description |
| ---- | ---- |
| `createdByPolicy` | When set to `true`, indicates that the controller created this object. This is important for pruning behavior. |
| `uid` | Used internally to track objects. |
| `diff` | Shows the differences between the policy `objectDefinition` and the actual cluster object. |
| `matchesAfterDryRun` | When set to `true`, indicates that an object initially did not match the policy, but a dry-run update produced a compliant result. The dry-run update can treat empty and null values as equivalent, so they are not reported as mismatches. This property may also be `true` if API server webhooks during a dry-run update produced a compliant object. |

> [!NOTE]

> > By default, the `diff` is hidden to protect sensitive data. To display the `diff` in the status, set the `recordDiff` field to `InStatus`.

Reviewer: The `diff` is only hidden by default for certain types (ConfigMaps, Secrets, and a few OpenShift resources like Routes).

```suggestion
> For some sensitive resources, the `diff` is hidden by default. To always display the `diff` in the status, set the `recordDiff` field on the `object-template` to `InStatus`.
```

For more information about templating, see the Configuration Policy Controller [Templating](#templating) section above.

#### Operator policy status details

After the policy evaluates, the `status` field contains the current compliance state and supporting details. A compliant `OperatorPolicy` matches the desired Operator configuration, and a noncompliant `OperatorPolicy` indicates one or more requirements are not met. For more info about noncompliant policies, see [Condition types](#condition-types) below.

> The following example shows a compliant `status` for an `OperatorPolicy` in `enforce` mode that installs the External Secrets Operator at version 0.11.0:

Reviewer: A full status example is quite long. Maybe putting it in an expandable section would be helpful:

```suggestion
<details>
<summary>This example shows a compliant `status` for an `OperatorPolicy` in `enforce` mode that installs the External Secrets Operator at version 0.11.0:</summary>

`` `yaml
status:
  ...
`` `

</details>
```

| `observedGeneration` | The generation of the `OperatorPolicy` resource when it was last observed |
| `conditions` | Detailed status conditions that track different aspects of `OperatorPolicy` compliance. For more information, see [Condition types](#condition-types). |
| `relatedObjects` | Kubernetes resources associated with the evaluated operator, such as `Subscription`, `ClusterServiceVersion`, `OperatorGroup`, and `Deployment` objects. |
| `resolvedSubscriptionLabel` | The resolved `name.namespace` of the `Subscription` resource |
| `overlappingPolicies` | List of other `OperatorPolicy` resources that manage the same subscription. Use this field to identify conflicting policies. |
| `subscriptionInterventionTime` | Timestamp indicating when the policy will intervene on a stuck subscription. A future timestamp means the controller is waiting for resolution. |

##### Condition types

The following condition types appear in a typical compliant status (see the example above). Additional conditions such as `MinorChannelUpgradeAvailable` may appear depending on compliance configuration. For more information, see [Compliance configuration](#compliance-configuration).

> - `ValidPolicySpec`: The policy spec is valid.

Reviewer: It might be nice to add some common ways the policy spec would not be valid, like invalid templates.

Author: Added some examples!

- `OperatorGroupCompliant`: The `OperatorGroup` matches what is required by the policy.
- `SubscriptionCompliant`: The `Subscription` matches what is required by the policy.
- `InstallPlanCompliant`: Installation and upgrade `InstallPlan` objects are either approved or waiting to be approved.
- `ClusterServiceVersionCompliant`: The installed operator version matches the `versions` list in `inform` mode, or meets policy requirements in `enforce` mode.
- `CustomResourceDefinitionCompliant`: Custom resource definitions (CRDs) required by the operator are present.
- `DeploymentCompliant`: Operator `Deployment` objects are running and available.
- `CatalogSourcesUnhealthy`: Reports `CatalogSource` health. When `status` is `"False"`, the catalog source was found and is healthy.
- `NoDeprecations`: The requested package, channel, and bundle are at recommended versions. When deprecations are detected, this condition reports them; whether that affects overall compliance is controlled by `complianceConfig.deprecationsPresent`.
- `Compliant`: Overall compliance summary aggregating the individual condition messages above.

##### Troubleshooting Tips

> When the policy spec is invalid, the condition `ValidPolicySpec` will appear `False` and the `status` will be `NonCompliant`. Below are some common error messages and suggested fixes.

Reviewer: These examples are great!
